src/integer/z/arithmetic/add.rs

```diff
@@ -398,3 +447,68 @@ mod test_add_between_z_and_zq {
         assert_eq!(e, Zq::from((0, i64::MAX)));
     }
 }
+
+#[cfg(test)]
+mod test_add_between_z_and_poly_over_z {
+    use super::Z;
+    use crate::integer::PolyOverZ;
+    use std::str::FromStr;
+
+    /// Testing addition for [`Z`] and [`PolyOverZ`]
+    #[test]
+    fn add() {
+        let a: Z = Z::from(9);
+        let b: PolyOverZ = PolyOverZ::from_str("4  1 1 2 3").unwrap();
+        let c: PolyOverZ = a + b;
+        assert_eq!(c, PolyOverZ::from_str("4  10 1 2 3").unwrap());
+    }
+
+    /// Testing addition for both borrowed [`Z`] and [`PolyOverZ`]
+    #[test]
+    fn add_borrow() {
+        let a: Z = Z::from(9);
+        let b: PolyOverZ = PolyOverZ::from_str("4  1 1 2 3").unwrap();
+        let c: PolyOverZ = &a + &b;
+        assert_eq!(c, PolyOverZ::from_str("4  10 1 2 3").unwrap());
+    }
+
+    /// Testing addition for borrowed [`Z`] and [`PolyOverZ`]
+    #[test]
+    fn add_first_borrowed() {
+        let a: Z = Z::from(9);
+        let b: PolyOverZ = PolyOverZ::from_str("4  1 1 2 3").unwrap();
+        let c: PolyOverZ = &a + b;
+        assert_eq!(c, PolyOverZ::from_str("4  10 1 2 3").unwrap());
+    }
+
+    /// Testing addition for [`Z`] and borrowed [`PolyOverZ`]
+    #[test]
+    fn add_second_borrowed() {
+        let a: Z = Z::from(9);
+        let b: PolyOverZ = PolyOverZ::from_str("4  1 1 2 3").unwrap();
+        let c: PolyOverZ = a + &b;
+        assert_eq!(c, PolyOverZ::from_str("4  10 1 2 3").unwrap());
+    }
+
+    /// Testing addition for large numbers
+    #[test]
+    fn add_large_numbers() {
+        let a: Z = Z::from(i64::MAX);
+        let b: PolyOverZ =
+            PolyOverZ::from_str(&format!("3  {} {} {}", i64::MAX, u64::MAX, i32::MAX)).unwrap();
+        let c: PolyOverZ = a + b;
+        assert_eq!(
+            c,
+            PolyOverZ::from_str(&format!("3  {} {} {}", u64::MAX - 1, u64::MAX, i32::MAX)).unwrap()
+        );
+    }
+
+    /// Testing addition for an empty polynomial
+    #[test]
+    fn add_zero_polynomial() {
+        let a: Z = Z::from(15);
+        let b: PolyOverZ = PolyOverZ::default();
+        let c: PolyOverZ = a + b;
+        assert_eq!(c, PolyOverZ::from_str("1  15").unwrap());
+    }
```

Comment:
I would also add the edgecase where you add a Z::ZERO (for consistency in both directions)

Reply:
added